cmd/lastfm.go

```diff
@@ -147,36 +156,51 @@ func getImageUrlForTrack(trackName string, artistName string, imageSize string)
 
 	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer res.Body.Close()
 
 	if res.StatusCode == http.StatusNotFound {
-		return "", ErrUserNotFound
+		return nil, errors.New("track not found")
 	}
 
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	var response GetTrackInfoResponse
 	err = json.Unmarshal([]byte(body), &response)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	for _, image := range response.Track.Album.Images {
 		if image.Size == imageSize {
-			return image.Link, nil
+			return &TrackInfo{response.Track.Album.AlbumName, image.Link}, nil
 		}
 	}
-	log.Println("No image found for track ", trackName, " and artist ", artistName, " and size ", imageSize)
-	return "", nil
+	return nil, errors.New("no image found")
 
 }
+
+func getImageUrlForArtist(artistUrl string) (string, error) {
+	url := artistUrl + "/+images"
+	log.Println("Getting image for artist ", url)
+	resp, err := http.Get(url)
+	if err != nil {
+		log.Println(err)
+	}
+	defer resp.Body.Close()
+
+	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return path.Base(doc.Find(".image-list-item-wrapper").First().Find("a").First().AttrOr("href", "")), nil
```

Comment:
Might be good to update this to capture potential errors if the DOM objects and whatnot aren't found. I suspect this would eventually happen if they change their frontend